src/pudl/transform/eia861.py

```diff
@@ -805,9 +835,14 @@ def transform(raw_dfs, eia861_tables=pc.pudl_tables["eia861"]):
         "balancing_authority_eia861": balancing_authority,
         "sales_eia861": sales,
         "demand_response_eia861": demand_response,
+        "distribution_systems_eia861": distribution_systems,
     }
     tfr_dfs = {}
 
+    # run early transform to change year to date and fix eia na cols.
+    for table_name, table in raw_dfs.items():
+        raw_dfs[table_name] = _early_transform(table)
+
```

Comment:
This should happen only after the check below, which ensures that there are actually some dataframes in the `raw_dfs` dictionary. I don't **think** there's any reason not to just put the early/pre transform step inside the very same loop that the main transformations are taking place within, is there? The one that starts with `for table in eia861_transform_functions`? That way all of the operations that are being run on the tables which are getting transformed are clearly outlined together.  It would also have avoided running the pre-transform step on things that aren't ready yet (b/c the metadata hasn't been compiled, etc.)

Also, does it make sense to be altering the raw dataframes in-place like this? Then they aren't really "raw" anymore.  I think in the `ferc714` version of this process I used the pre-transform process to populate the `tfr_dfs` dictionary of dataframes, and then had all of the transform functions depend only on `tfr_dfs` rather than both `raw_dfs` and `tfr_dfs`.

Reply:
That makes sense, will do.